### PR TITLE
Password login

### PR DESCRIPTION
**Pull request by T-S43**
Fellow team members, I'm done with this function if you could give me feedback it will be appreciated.
Thanks in advance.

Added:

- Password field in login
- New error page

Changed:

- Function sanitize is now in the database.
- Fixed an issue with login being replaced by student profile page

### CODE REVIEW

**Comment by joranvanerk** [comment]
Thank you for your contribution, everything is working now and we used the password verify function, and I think that's beautiful.

File: login-script.php [hunk 1]
```diff
@@ -4,33 +4,42 @@
...
+    // $pass = $sql = "SELECT * from `password` WHERE `passwd` = '$passwd'";
 
     $sql = "SELECT * from `password` WHERE `email` = '$email'";
+    // $passwdDB = "SELECT * from `password` WHERE `passwd` = '$passwd'";
+    
+    
     $result = mysqli_query($conn, $sql);
-
+    $record = mysqli_fetch_assoc($result);
+    $encpass = $record["passwd"];
```
Comment (zStevenn) [inline login-script.php:16]: With the registration process, there is also a salt created. A salt is a random 10 letter string, also saved in our database. You need this to verify the password as well.

File: login-script.php [hunk 2]
```diff
@@ -4,33 +4,42 @@
...
     if (mysqli_num_rows($result) == 1){
-        $record = mysqli_fetch_assoc($result);
-        echo("Successful login");
-        $sql = "SELECT * FROM `klant` WHERE `email` = '$email'";
-        $query = mysqli_query($conn, $sql);
-        $data = mysqli_fetch_assoc($query);
-        $userrole = "klant";
```
Comment (zStevenn) [inline login-script.php:18]: I believe we have multiple user roles, now every use gets assigned  the role 'klant'?

File: login-script.php [hunk 3]
```diff
@@ -4,33 +4,42 @@
...
-        $_SESSION["logged_in"] = true;
-        $_SESSION["userrole"] = $userrole;
-        // $_SESSION["voornaam"] = $data["voornaam"];
-        // $_SESSION["achternaam"] = $data["achternaam"];
-        setcookie("email", $record["email"], time()+3600);  /* expire in 1 hour */
-        setcookie("logged_in", true, time()+3600);  /* expire in 1 hour */
-        setcookie("userrole", $userrole, time()+3600);  /* expire in 1 hour */
-        // $_SESSION["rol"] = $record["rol"];
-        header("Location:./customer.php");
+        if(password_verify($passwd, $encpass)) {
```
Comment (zStevenn) [inline login-script.php:18]: if (password_verify($passwd.$salt, $encpass)) {success}

In the register process, the hashed password is created out of the following:

Passwd(user input)+salt(random 10 letter string)

**Review by zStevenn — CHANGES_REQUESTED** [review]
It looks like you have used all the correct functions to create the login-functionality! 

There are only a few minor changes that should be made, but those are only because the registration system is made a different way. If you change them it should be working perfectly!

**Comment by T-S43** [comment]
> It looks like you have used all the correct functions to create the login-functionality!
> 
> There are only a few minor changes that should be made, but those are only because the registration system is made a different way. If you change them it should be working perfectly!

Thanks for the feedback, I'll get right to it tomorrow

File: login-script.php (hunk 2 above)
Comment (T-S43) [reply login-script.php:18]: This branch is behind commits so this is a mini-scale task for the next sprint.

**Comment by joranvanerk** [comment]
Thank you for your contribution to this branch, we hope to see you in the future. You solved a conflict with the login script on line 12 and that's really clever, you used an redirection method that is very stable, thank you.

**Pull request merged**